### PR TITLE
Change iOS serialization from <string, string> to <string, object>

### PR DESCRIPTION
**Pull request by jwallra**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Previously, Xamarin.iOS mistakenly serialized all document values to strings.  It should try to serialize to the appropriate type.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #### iOS
 
 * **[Fix]** Fix a crash when tracking an exception without a message.
+* **[Fix]** Correcting document serialization to use the correct object type instead of always serializing to string.
```
Comment (guperrot) [inline CHANGELOG.md]:
```suggestion
* **[Fix]** Fix document serialization/deserialization to handle non string types inside an object.
```

File: Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs [hunk 2]
```diff
@@ -120,16 +122,56 @@ async void RunMBaaSAsync(object sender, EventArgs e)
...
+                        SomeObjectCollection = objectCollection,
+                        SomePrimitiveCollection = primitiveCollection
+                    }
                 };
                 var id = customDoc.Id.ToString();
                 var document = await Data.ReplaceAsync(id, customDoc, DefaultPartitions.UserDocuments);
                 AppCenterLog.Info(App.LogTag, "Replace result=" + JsonConvert.SerializeObject(document));
                 document = await Data.ReadAsync<CustomDocument>(id, DefaultPartitions.UserDocuments);
                 AppCenterLog.Info(App.LogTag, "Read result=" + JsonConvert.SerializeObject(document));
+                await Data.ReplaceAsync(Guid.NewGuid().ToString(), customDoc, DefaultPartitions.UserDocuments);
```
Comment (guperrot) [inline Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs]: Remove those 2 lines

File: Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs [hunk 3]
```diff
@@ -106,7 +108,7 @@ async void RunMBaaSAsync(object sender, EventArgs e)
                 var list = await Data.ListAsync<CustomDocument>(DefaultPartitions.UserDocuments);
                 foreach (var doc in list)
                 {
-                    AppCenterLog.Info(App.LogTag, "List result=" + JsonConvert.SerializeObject(doc));
+                    AppCenterLog.Info(App.LogTag, "List result=" + doc.Id);
```
Comment (guperrot) [inline Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs]: .JsonValue?

Comment (jwallra) [reply Apps/Contoso.Forms.Puppet/Contoso.Forms.Puppet/ModulePages/OthersContentPage.xaml.cs]: This is just a change I made to find doc Ids, I'll revert it so it prints everything again.

**Pull request merged**